src/arch/x86/x86_64/src/bzimage.rs

```diff
@@ -0,0 +1,293 @@
+            hd0_info: [0u8; 0x090 - 0x080],        // 0x080 /* obsolete! */
+            hd1_info: [0u8; 0x0a0 - 0x090],        // 0x090 /* obsolete! */
+            sys_desc_table: [0u8; 0x0b0 - 0x0a0],  // 0x0a0 /* obsolete! */
+            olpc_ofw_header: [0u8; 0x0c0 - 0x0b0], // 0x0b0
+            ext_ramdisk_image: 0u32,               // 0x0c0
+            ext_ramdisk_size: 0u32,                // 0x0c4
+            ext_cmd_line_ptr: 0u32,                // 0x0c8
+            _pad4: [0u8; 116],                     // 0x0cc
+            edid_info: [0u8; 0x1c0 - 0x140],       // 0x140
+            efi_info: [0u8; 0x1e0 - 0x1c0],        // 0x1c0
+            alt_mem_k: 0u32,                       // 0x1e0
+            scratch: 0u32,                         // 0x1e4 /* obsolete! */
+            e820_entries: 0u8,                     // 0x1e8
+            eddbuf_entries: 0u8,                   // 0x1e9
+            edd_mbr_sig_buf_entries: 0u8,          // 0x1ea
+            kbd_status: 0u8,                       // 0x1eb
+            secure_boot: 0u8,                      // 0x1ec
+            _pad5: [0u8; 2],                       // 0x1ed
+            sentinel: 0u8,                         // 0x1ef
+            _pad6: [0u8; 1],                       // 0x1f0
+            hdr: SetupHeader::default(),           // 0x1f1
+            _pad7: [0u8; 0x290 - 0x1f1 - size_of::<SetupHeader>()],
+            edd_mbr_sig_buffer: [0u32; 16],               // 0x290
+            e820_table: [E820Entry::default(); E820_MAX], // 0x2d0
+            _pad8: [0u8; 48],                             // 0xcd0
+            eddbuf: [0u8; 0xeec - 0xd00],                 // 0xd00
+            _pad9: [0u8; 276],                            // 0xeec
+        }
+    }
+}
+
+impl BzImage {
+    pub fn load(&mut self, w: &mut print::WriteTo) -> Result<usize> {
+        // The raw pointer shit is too painful.
+        let rom = SectionReader::new(&Memory {}, self.rom_base, self.rom_size);
+        let mut header: SetupHeader = {
+            let mut buff = [0u8; size_of::<SetupHeader>()];
+            rom.pread(&mut buff, HEADER_OFFSET).unwrap();
+            unsafe { transmute(buff) }
+        };
+
+        // first we make sure that the kernel is not too old
+
+        // from https://www.kernel.org/doc/Documentation/x86/boot.txt:
+        // For backwards compatibility, if the setup_sects field contains 0, the
+        // real value is 4.
+        if header.setup_sects == 0 {
+            header.setup_sects = 4;
+        }
+        // check magic numbers
+        if header.boot_flag != MAGIC_AA55 {
+            let i = header.boot_flag;
+            let p = unsafe { &header.boot_flag };
+            write!(w, "boot flag is {:x}, not {:x}, at {:x}, \n\r\n", i, MAGIC_AA55, p).unwrap();
+            return Err("magic number missing: header.boot_flag != 0xaa55");
+        }
+        if header.header != HDRS {
+            let i = header.header;
```

Comment:
Are these two lines necessary?

Reply:
I assume you mean 205 and 206? Yeah, it's useful for debugging for now. It can come out later.